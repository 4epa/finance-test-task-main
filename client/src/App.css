*{
  padding: 0;
  margin: 0;
  border: 0;
}

*,*:before,*:after{
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

:focus,:active{outline: none;}
a:focus,a:active{outline: none;}

nav,footer,header,aside{display:block;}

html,body{
  height: 100%;
  width: 100%;
  font-size: 100%;
  line-height: 1;
  font-size: 1rem;
  -ms-text-size-adjust: 100%;
  -moz-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

input,button,textarea{font-family: inherit;}

input::-ms-clear{display: none;}
button{cursor: pointer;}
button::-moz-focus-inner {padding: 0;border: 0;}
a,a:visited{text-decoration: none;}
a:hover{text-decoration: none;}
ul li {list-style: none;}
img{vertical-align: top;}

h1,h2,h3,h4,h5,h6{font-size: inherit;font-weight: 400;}

.header {
  padding: 15px;
  box-shadow: 1px 1px 5px #999;
  position: fixed;
  top: 0px;
  width: 100%;
  background-color: #fff;
  z-index: 10;
}
.header__container {
  max-width: 62.5rem;
  margin: 0 auto;
}
.logo {
  font-size: 1.563rem;
  font-weight: 700;
  color: rgb(86, 123, 202);
}
.content {
  padding: 50px 0px;
}