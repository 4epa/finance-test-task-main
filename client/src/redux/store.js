import { configureStore } from '@reduxjs/toolkit'
import sharesReducer from "./sharesSlice"

const store = configureStore({
  reducer: {
    shares: sharesReducer,
  },
})

export default store;

