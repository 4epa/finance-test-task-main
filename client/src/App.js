import "./App.css";
import { Route, Routes, BrowserRouter } from "react-router-dom";
import { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { getShares } from "./redux/selectors/sharesSelectors";
import { getSharesDataThunk, getTargetTicketThunk } from "./redux/sharesSlice";
import UserSharesList from "./pages/UserSharesTable/UserSharesList";
import { connectToServer, startShowQuote } from "./api/api";
import HomePage from "./pages/HomePage/HomePage";

const App = () => {
  const dispatch = useDispatch();

  const sharesList = useSelector((state) => getShares(state));
  const [isConnected, setIsConnected] = useState(false);
  const [showQuotes, setShowQuotes] = useState(false);

  // Connect to server and data request
  useEffect(() => {
    connectToServer();
    startShowQuote();
    setIsConnected(true);
    setShowQuotes(true);
    dispatch(getTargetTicketThunk());
  }, []);

  // Receiving data from the server
  useEffect(() => {
    if (showQuotes) {
      dispatch(getSharesDataThunk());
    }
  }, [showQuotes]);

  if (!isConnected) return <div>...Connect to server</div>;

  return (
    <BrowserRouter>
      <div className="App">
        <header className="header">
          <div className="header__container">
            <h1 className="logo">Financial analyst</h1>
          </div>
        </header>
        <main className="content">
          <Routes>
            <Route path="/" element={<HomePage sharesList={sharesList} />} />
            <Route
              path="/target-list"
              element={<UserSharesList sharesList={sharesList} />}
            />
          </Routes>
        </main>
      </div>
    </BrowserRouter>
  );
};

export default App;
