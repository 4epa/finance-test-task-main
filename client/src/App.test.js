import { render, screen } from "@testing-library/react";
import App from "./App";
import store from "./redux/store";
import { Provider } from "react-redux";

test("Render App component", () => {
  const component = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
