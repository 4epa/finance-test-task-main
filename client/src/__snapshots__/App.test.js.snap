// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="header"
        >
          <div
            class="header__container"
          >
            <h1
              class="logo"
            >
              Financial analyst
            </h1>
          </div>
        </header>
        <main
          class="content"
        >
          <div>
            <div
              class="breadcrumb"
              role="presentation"
            >
              <nav
                aria-label="breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-a9lb73-MuiTypography-root"
                    >
                      Main page
                    </p>
                  </li>
                </ol>
              </nav>
            </div>
            <h2
              class="title"
            >
              You may be interested
            </h2>
            <div
              class="section"
            >
              <div
                class="table"
              >
                <header
                  class="table_header"
                >
                  <div
                    class="header_item"
                  >
                    <button
                      class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span>
                        ticker
                      </span>
                      <span />
                    </button>
                  </div>
                  <div
                    class="header_item"
                  >
                    <button
                      class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span>
                        name
                      </span>
                      <span />
                    </button>
                  </div>
                  <div
                    class="header_item"
                  >
                    <button
                      class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span>
                        price
                      </span>
                      <span />
                    </button>
                  </div>
                  <div
                    class="header_item"
                  >
                    <button
                      class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span>
                        percent
                      </span>
                      <span />
                    </button>
                  </div>
                  <div
                    class="header_item"
                  >
                    <button
                      class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span>
                        dividend
                      </span>
                      <span />
                    </button>
                  </div>
                </header>
                <div
                  class="list"
                >
                  <div
                    class="container"
                  >
                    <div
                      class="img"
                    >
                      <img
                        alt="list Empty"
                        src="empty_list.png"
                      />
                    </div>
                    <p
                      class="text"
                    >
                      There is nothing here yet
                    </p>
                  </div>
                </div>
                <footer
                  class="footer"
                >
                  <a
                    class="list_link"
                    href="/target-list"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-wt4i4k-MuiSvgIcon-root"
                      data-testid="ListIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
                      />
                    </svg>
                    <span>
                      View list
                    </span>
                    <span>
                      0
                    </span>
                  </a>
                </footer>
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="header"
      >
        <div
          class="header__container"
        >
          <h1
            class="logo"
          >
            Financial analyst
          </h1>
        </div>
      </header>
      <main
        class="content"
      >
        <div>
          <div
            class="breadcrumb"
            role="presentation"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
            >
              <ol
                class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-a9lb73-MuiTypography-root"
                  >
                    Main page
                  </p>
                </li>
              </ol>
            </nav>
          </div>
          <h2
            class="title"
          >
            You may be interested
          </h2>
          <div
            class="section"
          >
            <div
              class="table"
            >
              <header
                class="table_header"
              >
                <div
                  class="header_item"
                >
                  <button
                    class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span>
                      ticker
                    </span>
                    <span />
                  </button>
                </div>
                <div
                  class="header_item"
                >
                  <button
                    class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span>
                      name
                    </span>
                    <span />
                  </button>
                </div>
                <div
                  class="header_item"
                >
                  <button
                    class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span>
                      price
                    </span>
                    <span />
                  </button>
                </div>
                <div
                  class="header_item"
                >
                  <button
                    class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span>
                      percent
                    </span>
                    <span />
                  </button>
                </div>
                <div
                  class="header_item"
                >
                  <button
                    class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span>
                      dividend
                    </span>
                    <span />
                  </button>
                </div>
              </header>
              <div
                class="list"
              >
                <div
                  class="container"
                >
                  <div
                    class="img"
                  >
                    <img
                      alt="list Empty"
                      src="empty_list.png"
                    />
                  </div>
                  <p
                    class="text"
                  >
                    There is nothing here yet
                  </p>
                </div>
              </div>
              <footer
                class="footer"
              >
                <a
                  class="list_link"
                  href="/target-list"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-wt4i4k-MuiSvgIcon-root"
                    data-testid="ListIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
                    />
                  </svg>
                  <span>
                    View list
                  </span>
                  <span>
                    0
                  </span>
                </a>
              </footer>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
